deprecate `ZArith.Zdigits`

andres-erbsen opened the pull request:
- [x] Added **changelog**.

andres-erbsen commented:
@coqbot run full ci

Alizter commented on doc/changelog/11-standard-library/17733-deprecate-Zdigits.rst:
```diff
@@ -0,0 +1,4 @@
+- **Deprecated:** :g:`ZArith.Zdigits`
+  Describe your change here but do not end with a period
```
needs updating

Alizter reviewed (commented):
could you add what to use instead for the deprecations?

andres-erbsen commented:
> could you add what to use instead for the deprecations? i.e. using the note option of the deprecated attribute.

I added `note=` statements. The standard library does not have an exact drop-in replacement for this functionality, so some of the recommendations are based on anticipated context of use. I added a corresponding request for comments to the changelog.

andres-erbsen commented:
Full CI failures at https://github.com/coq/coq/commit/19123b441c7ce6f54ac7b45d041e9898fdd83f3f look unrelated (geocoq and ltac2 compiler)

gares commented:
ping @coq/number-maintainers this PR seems ready, can you review it quickly please?

proux01 reviewed (approved):
Sorry, I thought I already merged that one (I apparently mixed it with https://github.com/coq/coq/pull/17601 ), let's merge.

proux01 commented:
@coqbot merge now
